extensions/sql-database-projects/src/models/project.ts

```diff
@@ -10,6 +10,8 @@ import * as utils from '../common/utils';
 import * as xmlFormat from 'xml-formatter';
 import * as os from 'os';
 import * as UUID from 'vscode-languageclient/lib/utils/uuid';
+import type * as azdataType from 'azdata';
```

Comment:
a normal `import * as azdata from 'azdata';` doesn't work here?

Reply:
this is done throughout the extension so that the import doesn't break the vscode extension when it can't find azdata. Need to remember to go back here and add support for getting the SqlProjectsService from vscode-mssql. Similar to what we do for all these calls in projectController: https://github.com/microsoft/azuredatastudio/blob/bc85f52dd45852b86dcecabe5ff6cc68f2f0711e/extensions/sql-database-projects/src/controllers/projectController.ts#L518-L523

Reply:
For further context - doing `import * as azdata from 'azdata'` tries to import the module named `azdata` as soon as this module (file) is loaded. But in VS Code this doesn't exist, and so it will throw an exception and the extension will fail to load.

`import type *` tells TS that this is JUST importing the typings and so is a compile-only thing, it is removed in the actual generated JS and thus won't have that problem.